src/Athena/Controllers/api/ProgramController.cs

```diff
@@ -0,0 +1,88 @@
+    {
+        private readonly IProgramRepository programs;
+        private readonly IInstitutionRepository institutions;
+        private readonly IStudentRepository students;
+        private readonly IRequirementRepository requirements;
+        
+        public ProgramController(IProgramRepository programsRepository, IInstitutionRepository institutionRepository, IStudentRepository studentRepository, IRequirementRepository requirementRepository)
+        {
+            programs = programsRepository ?? throw new ArgumentNullException(nameof(programsRepository));
+            institutions = institutionRepository ?? throw new ArgumentNullException(nameof(institutionRepository));
+            students = studentRepository ?? throw new ArgumentNullException(nameof(studentRepository));
+            requirements = requirementRepository ?? throw new ArgumentNullException(nameof(requirementRepository));
+        }
+
+        [HttpPost]
+        public async Task AddProgram([FromBody] Program program) => await programs.AddAsync(program);
+
+        [HttpGet("{id}")]
+        public async Task<Program> GetProgram(Guid id) => await programs.GetAsync(id);
+
+        [HttpPut("{id}")]
+        public async Task EditProgram (Guid id, [FromBody] Program program)
+        {
+            if (id != program.Id)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to edit {id} but got a model for {program.Id}");
+            }
+            await programs.EditAsync(program);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task DeleteProgram(Guid id)
+        {
+            var program = await programs.GetAsync(id);
+            if (program == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to delete {program} that does not exist");
+            }
+            await programs.DeleteAsync(program);
+        }
+
+        [HttpGet("{id}/requirements")]
+        public async Task<IEnumerable<Requirement>> GetRequirementsForProgramAsync(Guid id)
+        {
+            var program = await programs.GetAsync(id);
+            if (program == null)
+            {
+                throw new ApiException(HttpStatusCode.NotFound, $"program with id {id} not found");
+            }
+            return (await requirements.GetRequirementsForProgramAsync(program));
+        }
+
+        [HttpPost("api/v1/student/{id}/programs/{programId}")]
+        public async Task RegisterStudentForProgram(Student student, Program program)
+        {
+            if (student == null || program == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to register for program {program}, where the student {student} or program {program} does not exist");
+            }
+            await students.RegisterForProgramAsync(student, program);
+        }
+
+        [HttpDelete("api/v1/student/{id}/programs/{programId}")]
```

Comment:
Did you intend to implement this here or in a composite controller? If you intend to implement it here you need to use `/api/v1/student...` to override the controller-wide prefix.